src/lib/sql/sql_translator.cpp

```diff
@@ -355,11 +356,21 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
+  std::vector<const hsql::Expr*> condition_list;
```

Comment:
One more thing. Somewhere in this method we should have a comment along the lines of „We cannot handle AND/OR in joins yet. However, in some cases, part of the AND condition is not relevant for the join and is actually a predicate for one of the input sides. We identify these cases and push those predicate down to the correct side.“